src/Language/Haskell/Exts/Pretty.hs

```diff
@@ -490,8 +490,10 @@ instance  Pretty (Decl l) where
 
         pretty (FunBind _ matches) = do
                 e <- fmap layout getPPEnv
-                case e of PPOffsideRule -> foldr ($$$) empty (map pretty matches)
-                          _ -> foldr (\x y -> x <> semi <> y) empty (map pretty matches)
+                case e of
+                    PPOffsideRule -> foldr ($$$) empty (map pretty matches)
+                    PPInLine -> foldr ($$$) empty (map pretty matches)
```

Comment:
I don't think this fixes the problem properly. The `PPInLine` option is meant to print

```
foo a = 1
foo b = 2
```
as

```
foo a = 1; foo b = 2
```

which is not what happens with this patch.

The correct fix is to change line 494 from

```
_ -> foldr (\x y -> x <> semi <> y) empty (map pretty matches)
```
to
```
_ -> hsep $ punctuate semi (map pretty matches)
```

which will deal with the `n=0` case correctly.